File: app/src/main/java/com/selflearning/chemistree/chemistry/inorganic/acids/AcidsDao.java
package com.selflearning.chemistree.chemistry.inorganic.acids;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface AcidsDao {
    @Query("SELECT * FROM acids")
    List<Acids> getAcids();

    @Query("SELECT * FROM acids WHERE difficult < :lvl")
    List<Acids> getAcidsByLevel(int lvl);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<Acids> acids);
}

File: app/src/main/java/com/selflearning/chemistree/chemistry/inorganic/acids/AcidsDao.kt
package com.selflearning.chemistree.chemistry.inorganic.acids

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface AcidsDao {
    @Query("SELECT * FROM acids")
    fun getAll(): Flow<List<Acid>>

    @Query("SELECT * FROM acids WHERE difficult < :lvl")
    fun getAcidsByLevel(lvl: Int): List<Acid>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(acids: List<Acid>)
}